src/application/service_tester.go

```diff
@@ -141,11 +144,46 @@ func (s *ServiceTester) setup() error {
 
 // Run runs the tests for the service and return true if the tests passed
 // and an error if any
-func (s *ServiceTester) Run() (int, error) {
+func (s *ServiceTester) Run() (types.TestResult, error) {
 	if err := s.setup(); err != nil {
-		return 1, err
+		return types.TestResult{
+			Passed:      false,
+			Interrupted: false,
+		}, err
```

Comment:
Since you are returning an error the values of passed/interrupted don't matter. This could be 
```
return types.TestResult{}, err
```
Please apply this in similar places elsewhere